pkg/util/sem/sem.go

```diff
@@ -66,26 +69,90 @@ const (
 
 var (
 	semEnabled int32
+	// sysMap record original value of sysvar
+	sysMap      map[string]string
+	mapMutex    sync.Mutex
+	sysMapMutex sync.RWMutex
 )
 
+var isResVarAdmin int32 = 0
+
+// CheckResVarAdmin checks if the user has the RESTRICTED_VARIABLES_ADMIN privilege
+func CheckResVarAdmin(sctx sessionctx.Context) {
```

Comment:
Why this is not returning a boolean, instead set a variable of package scope? 
The input value sctx can be arbitrary, so the value `isResVarAdmin` is not stable.